src/ConfigMaps.ts

```diff
@@ -0,0 +1,231 @@
+		await createSecretFromUri(uri);
+	});
+
+}
+
+async function createConfigMapFromUri(uri:vscode.Uri) {
+	await createConfigMapFromFilenameOrFolder(uri)
+		.then ( (output) => {
+			extension.setStatusLineMessage(`Received... ${output}`);
+		}).catch( (error) => {
+			utils.shareMessage(extension.mainOutputChannel, ("Error encountered while creating config-map: " + error));
+	});
+	extension.hideStatusLine();
+}
+
+async function createSecretFromUri(uri:vscode.Uri) {
+	await createSecretFromFilenameOrFolder(uri)
+		.then ( (output) => {
+			extension.setStatusLineMessage(`Received... ${output}`);
+		}).catch( (error) => {
+			utils.shareMessage(extension.mainOutputChannel, ("Error encountered while creating secret: " + error));
+	});
+	extension.hideStatusLine();
+}
+
+async function createConfigMapFromFilenameOrFolder(uri:vscode.Uri) : Promise<any> {
+	return new Promise( async (resolve, reject) => {
+		const kubeIsReady = await isKubernetesAvailable();
+		if (kubeIsReady && kubeIsReady === true) {
+			let configMapName = await vscode.window.showInputBox({
+				prompt: 'Config Map Name',
+				placeHolder: 'Provide the unique identifier for this Kubernetes config map.',
+				value: 'new-config-map',
+				ignoreFocusOut: true
+			});
+			if (configMapName) {
+				createConfigMap(configMapName, uri)
+					.then( (output) => {
+						vscode.window.showInformationMessage(`Successfully created new config-map named "${configMapName}"`);
+						resolve(output);
+					})
+					.catch(err => {
+						vscode.window.showErrorMessage(`Problem encountered creating new config-map named "${configMapName}": ${err}`);
+						reject(err);
+					});
+			} else {
+				reject("No name given for config map.");
+			}
+		} else {
+			vscode.window.showInformationMessage(`Kubernetes extensions still coming up, please wait a moment and try again.`);
+		}
+	}); 
+}
+
+async function createSecretFromFilenameOrFolder(uri:vscode.Uri) : Promise<any> {
+	return new Promise( async (resolve, reject) => {
+		const kubeIsReady = await isKubernetesAvailable();
+		if (kubeIsReady && kubeIsReady === true) {
+			let secretName = await vscode.window.showInputBox({
+				prompt: 'Secret Name',
+				placeHolder: 'Provide the unique identifier for this Kubernetes secret.',
+				value: 'new-secret',
+				ignoreFocusOut: true
+			});
+			if (secretName) {
+				createSecret(secretName, uri)
+					.then( (output) => {
+						vscode.window.showInformationMessage(`Successfully created new secret named "${secretName}"`);
+						resolve(output);
+					})
+					.catch(err => {
+						vscode.window.showErrorMessage(`Problem encountered creating new secret named "${secretName}": ${err}`);
+						reject(err);
+					});
+			} else {
+				reject("No name given for secret.");
+			}
+		} else {
+			vscode.window.showInformationMessage(`Kubernetes extensions still coming up, please wait a moment and try again.`);
+		}
+	}); 
+}
+
+async function isKubernetesAvailable(): Promise<any> {
```

Comment:
I think the method is nto checking if Kubernetes is available but if Camel is available inside Kubernetes (which requires Kubernetes to be available)

Reply:
Will spin the camel check into another utility method somewhere else and just rely on kubectl.available for this one.